assets/src/edit-story/app/prepublish/components/videoOptimization.js

```diff
@@ -32,14 +40,46 @@ const Thumbnail = styled.img`
   object-position: 50% 50%;
 `;
 
+const OptimizeButton = styled(Button)`
+  margin: auto 16px;
+  border: ${({ theme }) => `1px solid ${theme.colors.border.defaultNormal}`};
+`;
 export function VideoOptimization({ element }) {
+  const { resource = {} } = element;
+  const { optimizeVideo } = useLocalMedia((state) => ({
+    optimizeVideo: state.actions.optimizeVideo,
+  }));
+
+  const optimizingVideoRef = useRef();
+
+  const handleUpdateVideo = useCallback(async () => {
+    optimizingVideoRef.current = true;
+    // If optimization fails we want to update the text
+    // temporary solution until more robust error handling is ready
+
+    await optimizeVideo({ resource }).finally(() => {
+      // TODO add in an error state
+      optimizingVideoRef.current = false;
+    });
+  }, [resource, optimizeVideo]);
+
   return (
     <Container>
       <Thumbnail
-        src={element.resource?.poster}
-        alt={element.resource?.alt || __('video thumbnail', 'web-stories')}
+        src={resource?.poster}
+        alt={resource?.alt || __('video thumbnail', 'web-stories')}
         crossOrigin="anonymous"
       />
+      <OptimizeButton
+        type={BUTTON_TYPES.TERTIARY}
+        size={BUTTON_SIZES.SMALL}
+        onClick={handleUpdateVideo}
+        disabled={optimizingVideoRef?.current}
+      >
+        {optimizingVideoRef?.current
```

Comment:
what was the reason behind making this a ref? I feel like since it affects the rendering of components it should just be state. Thoughts?

Reply:
nope state makes more sense, i was doing this kinda late - i blame my brain.